### PR TITLE
[GeneratorBundle] refactored page generator

### PR DESCRIPTION
**Pull request by deZinc**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets |

### CODE REVIEW

File: src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php [hunk 1]
```diff
@@ -342,13 +346,14 @@ protected function askForSections(
...
-    protected function getAvailableSections(BundleInterface $bundle, $context = null, $defaultSections = array())
+    protected function getAvailableSections(BundleInterface $bundle, $context = null, $defaultSections = [])
     {
-        $configs = array();
+        $configs = [];
         $counter = 1;
 
         // Get the available sections from disc
-        $dir = Kernel::VERSION_ID >= 40000 ? $this->getContainer()->getParameter('kernel.project_dir') . '/config/kunstmaancms/pageparts/' : $bundle->getPath() . '/Resources/config/pageparts/';
+        $dir = Kernel::VERSION_ID >= 40000 ? $this->getContainer()->getParameter('kernel.project_dir'
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php]: Revert this change, makes the line more unreadable

File: src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php [hunk 2]
```diff
@@ -852,11 +888,16 @@ protected function getEntityFields(
...
     protected function getAvailableTemplates(BundleInterface $bundle)
     {
-        $configs = array();
+        $configs = [];
         $counter = 1;
 
         // Get the available sections from disc
-        $dir = $bundle->getPath().'/Resources/config/pagetemplates/';
+        if (Kernel::VERSION_ID >= 40000) {
+            $dir = $bundle->getPath().'/../config/kunstmaancms/pagetemplates/';
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php]:
```suggestion
                $dir = $this->container->getParameter('kernel.project_dir') . '/config/kunstmaancms/pagetemplates/';
```

To keep the code inline with the check in [`KunstmaanGenerateCommand`](https://github.com/Kunstmaan/KunstmaanBundlesCMS/blob/7b48b9c1d0bcecca7970011e8ef46d818d3a9ebb/src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php#L351)

File: src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php [hunk 3]
```diff
@@ -218,7 +218,12 @@ protected function generateEntityAdminType(
     protected function installDefaultPageTemplates($bundle)
     {
         // Configuration templates
-        $dirPath = sprintf('%s/Resources/config/pagetemplates/', $bundle->getPath());
+        if (Kernel::VERSION_ID >= 40000) {
+            $dirPath = sprintf('%s/../config/kunstmaancms/pagetemplates/', $bundle->getPath());
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php]:
```suggestion
                $dirPath = $this->container->getParameter('kernel.project_dir') . '/config/kunstmaancms/pagetemplates/';
```

To keep the code inline with the check in [`KunstmaanGenerateCommand`](https://github.com/Kunstmaan/KunstmaanBundlesCMS/blob/7b48b9c1d0bcecca7970011e8ef46d818d3a9ebb/src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php#L351)

File: src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php [hunk 4]
```diff
@@ -236,7 +241,12 @@ protected function installDefaultPageTemplates($bundle)
         }
 
         // Twig templates
-        $dirPath = sprintf('%s/Resources/views/Pages/Common/', $bundle->getPath());
+        if (Kernel::VERSION_ID >= 40000) {
+            $dirPath = sprintf('%s/../templates/Pages/Common/', $bundle->getPath());
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php]: The whole if/else can be simplified to this one line:
```suggestion
            $dirPath = $this->getTemplateDir($this->bundle) . '/Pages/Common/'';
```

File: src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php [hunk 5]
```diff
@@ -269,7 +279,12 @@ protected function installDefaultPageTemplates($bundle)
     protected function installDefaultPagePartConfiguration($bundle)
     {
         // Pagepart configuration
-        $dirPath = sprintf('%s/Resources/config/pageparts/', $bundle->getPath());
+        if (Kernel::VERSION_ID >= 40000) {
+            $dirPath = sprintf('%s/../config/kunstmaancms/pageparts/', $bundle->getPath());
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php]:
```suggestion
                $dirPath = $this->container->getParameter('kernel.project_dir') . '/config/kunstmaancms/pageparts/';
```

To keep the code inline with the check in [`KunstmaanGenerateCommand`](https://github.com/Kunstmaan/KunstmaanBundlesCMS/blob/7b48b9c1d0bcecca7970011e8ef46d818d3a9ebb/src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php#L351)

File: src/Kunstmaan/GeneratorBundle/Generator/PageGenerator.php [hunk 6]
```diff
@@ -179,7 +180,7 @@ private function updateParentPages()
         foreach ($this->parentPages as $file) {
             $data = file_get_contents($file);
             $data = preg_replace(
-                '/(function\s*getPossibleChildTypes\s*\(\)\s*\{\s*return\s*array\s*\()/',
+                '/(function\s*getPossibleChildTypes\s*\(\)\s*\{\s*return\s*\[)/',
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Generator/PageGenerator.php]: Should we still allow `array` and `[` to keep BC with older classes?

File: src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/page/Entity/Pages/ExtraFunctions.php [hunk 7]
```diff
@@ -45,5 +43,5 @@ public function getPageTemplates()
      */
     public function getDefaultView()
     {
-        return '{{ bundle }}:Pages:Common/view.html.twig';
+        return '{% if not isV4 %}{{ bundle.getName() }}/{%endif%}Pages/Common/view.html.twig';
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/page/Entity/Pages/ExtraFunctions.php]:
```suggestion
        return '{% if not isV4 %}{{ bundle.getName() }}:{%endif%}Pages/Common/view.html.twig';
```

There should be a ":" after the bundlename, not a /

**Comment by acrobat** [comment]
@deZinc Also there is a style issue now, see the styleci check

File: src/Kunstmaan/GeneratorBundle/Generator/PageGenerator.php (hunk 6 above)
Comment (deZinc) [reply src/Kunstmaan/GeneratorBundle/Generator/PageGenerator.php]: you're right, that would be better

File: src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php [hunk 8]
```diff
@@ -255,7 +261,7 @@ protected function installDefaultPageTemplates($bundle)
         $contents = file_get_contents($dirPath . 'view.html.twig');
         if (strpos($contents, '{% extends ') === false) {
             GeneratorUtils::prepend(
-                "{% extends '" . $bundle->getName() . ":Layout:layout.html.twig' %}\n",
+                "{% extends '" . $bundle->getName() . "/Layout/layout.html.twig' %}\n",
```
Comment (acrobat) [inline src/Kunstmaan/GeneratorBundle/Generator/KunstmaanGenerator.php]: There should a symfony 4 check here as `App/Layout/layout.html.twig` doesn't work.

SF4:
```
"{% extends 'Layout/layout.html.twig' %}\n",
```

SF3.4
```
"{% extends '" . $bundle->getName() . ":Layout:layout.html.twig' %}\n",
```

File: src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php (hunk 2 above)
Comment (acrobat) [reply src/Kunstmaan/GeneratorBundle/Command/KunstmaanGenerateCommand.php]: Comment above is still todo

**Comment by acrobat** [comment]
@deZinc can you rebase the pr to fix the merge conflict + check the styleci error. Thanks!
